src/priceClient/priceClient.e2e.ts

```diff
@@ -69,6 +87,17 @@ function validateTokenPrice(tokenPrice: TokenPrice, address: string, timestamp:
   );
 }
 
+describe("PriceClient: BaseHTTPAdapter", function () {
+  test("Retry behaviour", async function () {
```

Comment:
nit: it might be nice to have a test that ensures the retry process succeeds and returns the right result if some call before nRetries is successful. Basically the opposite of this test.

Reply:
This will require some introspection with axios but it's a good point - we should test this. I'll work on adding that.

Reply:
Test added here: https://github.com/across-protocol/sdk-v2/pull/157/commits/5c42d1b17c93842f98587011174017013f94b4f9. It works by permitting the request to finally succeed on the final retry.